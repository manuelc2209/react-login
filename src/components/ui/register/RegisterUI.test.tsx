import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { RegisterUI } from '../..';

const buttonProps = {
    label: 'Go back',
    disabled: true,
    onClick: () => jest.fn()
};

const formProps = {
    name: 'Test',
    disabled: false,
    validatePassword: true,
    doubleValidation: true,
    passwordLabel: 'PwLabel',
    nicknamePlaceholder: 'Nickname Placeholder',
    passwordPlaceholder: 'Pw Placeholder',
    onClick: () => jest.fn()
};

it('renders snapshot correctly', () => {
    const tree = renderer
        .create(<RegisterUI buttonProperties={buttonProps} formProperties={formProps} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RegisterUI buttonProperties={buttonProps} formProperties={formProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
});
