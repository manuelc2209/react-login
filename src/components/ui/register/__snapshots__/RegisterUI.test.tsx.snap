// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<div
  className="sc-jJoQJp cbKpxQ"
>
  <div
    className="sc-pVTFL jvgrHT sc-jcFjpl NIgNM"
  >
    <button
      className="sc-gsDKAQ hTByPO"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      size="small"
    >
      <span
        className="sc-bdvvtL jvqAQd"
      >
        Go back
      </span>
    </button>
  </div>
  <div
    className="sc-cCcXHH mvRgQ"
  >
    <div
      className="sc-cidDSM kDJmsM"
    >
      <div
        className="sc-hGPBjI hOLmWg sc-hiCibw"
        disabled={false}
      >
        <div
          className="sc-hGPBjI hOLmWg"
        >
          <div
            className="sc-iqseJM jIljjx"
          >
            <span
              className="sc-kDTinF"
            >
              Test
            </span>
            <input
              className="sc-jrQzAO eGrRgi"
              disabled={false}
              onChange={[Function]}
              placeholder="Nickname Placeholder"
            />
          </div>
        </div>
        <div
          className="sc-fotOHu fpfbUY"
        >
          <div
            className="sc-fFeiMQ gdSowf"
          >
            <div
              className="sc-iqseJM jIljjx"
            >
              <span
                className="sc-kDTinF"
              >
                PwLabel
              </span>
              <input
                className="sc-jrQzAO eGrRgi"
                disabled={false}
                onChange={[Function]}
                placeholder="Pw Placeholder"
                type="password"
              />
            </div>
            <span
              className="sc-bkkeKt dskASR"
            />
          </div>
          <button
            className="sc-gsDKAQ hDLjCu sc-dJjYzT jXkYNx"
            disabled={true}
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            size="large"
          >
            <span
              className="sc-bdvvtL jvqAQd"
              disabled={true}
            >
              Display Password
            </span>
          </button>
        </div>
        <div
          className="sc-iqseJM jIljjx"
        >
          <span
            className="sc-kDTinF"
          >
            Repeat PwLabel
          </span>
          <input
            className="sc-jrQzAO eGrRgi"
            disabled={false}
            onChange={[Function]}
            placeholder="Please repeat your PwLabel"
            type="password"
          />
        </div>
        <button
          className="sc-gsDKAQ joOHgp sc-ieecCq"
          disabled={true}
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          size="small"
        >
          <span
            className="sc-bdvvtL jvqAQd"
            disabled={true}
          >
            Submit
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
